### DISCUSSION TITLE
mh_z19 returns {}

### DISCUSSION THREAD

**Discussion by Sudharma1997** [UedaTakeyuki/mh-z19#56] [Q&A]
Despite meticulously setting all permissions, diligently following the setup.sh script, and ensuring all prerequisites are met, we are still encountering an empty response ({}) when executing python3 -m mh_z19.

**Comment by UedaTakeyuki** [comment]
The return {} means that the sensor module didn't respond.
- The sensor module needs to pre-heat, or not respond.
- The sensor Module is broken.


In addition to connecting via serial, also [PWM](https://github.com/UedaTakeyuki/mh-z19/wiki/PWM-support.) is available, so please try it.
